docs/plugins.md

```diff
@@ -0,0 +1,352 @@
+# Plugin Handbook
+
+## Table of Contents
+
+1. [Introduction](#introduction)
+   1. [What is a Lighthouse Plugin?](#what-is-a-lighthouse-plugin)
+   1. [Plugins v. Custom Config](#plugins-v-custom-config)
+   1. [Getting Started](#getting-started)
+1. [API](#api)
+   1. [Plugin Config](#plugin-config)
+   1. [Plugin Audits](#plugin-audits)
+1. [Best Practices](#best-practices)
+   1. [Naming](#naming)
+   1. [Scoring](#scoring)
+   1. [Common Mistakes](#common-mistakes)
+
+## Introduction
+
+If you're new to Lighthouse development, start by reading up on the overall [architecture](./architecture.md) and how [configuration](./configuration.md) works before continuing.
+
+### What is a Lighthouse Plugin?
+
+Lighthouse plugins are a way to extend the functionality of Lighthouse with insight from domain experts (that's you!) and easily share this extra functionality with other Lighthouse users. At its core, a plugin is a node module that implements a set of checks that will get run by Lighthouse and added to the report as a new category.
+
+### Do You Need a Custom Audit?
```

Comment:
i dont really understand this section. mostly it feels kinda unnecessary. i have problems with the content too.. so maybe you'll just agree it's not important and we can ignore all that. :)

Reply:
haha, fair points! My dangling sentence was going to say "consider the criteria for Lighthouse audits when adding your own" (i.e. it's actionable, it's impactful, etc) but if we feel like this is getting too preachy that plugin devs know what they're doing, I understand.

Reply:
(removed it already but wanted to leave the point here for thought)